lib/wor/authentication.rb

```diff
@@ -1,45 +1,57 @@
+require_relative 'authentication/exceptions'
 require_relative 'authentication/controller'
 require_relative 'authentication/decoded_token'
-require_relative 'authentication/exceptions'
 require_relative 'authentication/sessions_controller'
 require_relative 'authentication/token_manager'
 require_relative 'authentication/version'
 
 module Wor
   module Authentication
     @config = {
-      expiration_days: 2,
-      maximum_useful_days: 30
+      token_key: nil,
+      expiration_minutes: 43200,
+      renew_minutes: 43185
     }
 
     def self.configure
       yield self
+      if renew_minutes >= expiration_minutes
+        raise Wor::Authentication::Exceptions::InvalidMaximumUsefulDays
+      end
     end
 
-    def self.expiration_days=(expiration_days)
-      unless expiration_days.is_a? Integer
-        raise Wor::Authentication::Exceptions::InvalidExpirationDaysError
+    def self.expiration_minutes=(expiration_minutes)
+      unless expiration_minutes.is_a? Integer
+        raise Wor::Authentication::Exceptions::InvalidDateType
```

Comment:
Why is this InvalidDateType if it is an integer ?

Reply:
It is kind of a convention to represent timestamps as integers, also we compare dates as ints in the validation of the token.

Reply:
Yes, Dates are represented in integers, but not when that integer represents minutes

Reply:
Any updates on this correction ?